bot/__init__.py

```diff
@@ -168,6 +171,46 @@ def _get_word(self) -> str:
         result = self.buffer[self.previous:self.index + (pos+2)]
         self.index += 2
 
+    # Check if a command in the form of `bot.tags['ask'] {= 'whatever'}` was used
+    elif current == "[":
+        # Helper function to remove any characters we don't care about
+        def clean_argument(arg: str) -> str:
+            return arg.strip("[]'\" ")
+
+        # Syntax is `bot.tags['ask']` => mimic `getattr`
+        if self.buffer.endswith("]"):
+            # Key: The first argument, specified `bot.tags[here]`
+            key = clean_argument(self.buffer[self.index:])
+            # note: if not key, this corresponds to an empty argument
+            #       so this should throw / return a SyntaxError ?
+            args = f"\"{key}\""
+            # Use the cog's `get` command.
+            result = self.buffer[self.previous:self.index] + ".get"
+
+        # Syntax is `bot.tags['ask'] = 'whatever'` => mimic `setattr`
+        elif "=" in self.buffer and not self.buffer.endswith("="):
+            equals_pos = self.buffer.find("=")
+            # Key: The first argument, specified `bot.tags[here]`
+            key = clean_argument(self.buffer[self.index:equals_pos])
+            # Value: The second argument, specified after the `=`
+            value = (clean_argument(self.buffer.split("=")[1])
+                     .replace("\"", "\\\"")  # escape any unescaped quotes
+                     .replace("'", "\\'"))   # to mimic triple quote behaviour.
+            # Use the cog's `set` command.
+            result = self.buffer[self.previous:self.index] + ".set"
+            args = f"\"{key}\" \"{value}\""
+
+        # Syntax is god knows what, pass it along
+        # in the future, this should probably return / throw SyntaxError
+        else:
+            result = self.buffer
+            args = ''
+
+        # Reconstruct valid discord.py syntax
+        self.buffer = f"{result} {args}"
+        self.index = len(result)
+        self.end = len(self.buffer)
```

Comment:
this is being repeated in the () and [] handling. Can the code be refactored so this happens outside the if else block?

Reply:
i actually don't know where this is being repeated, I don't handle a special case for `[]` (it will just get passed as `bot tags get ""` so the "empty thing), and not sure where `()` is handled. Can you elaborate?

Reply:
I mean I basically do this in the regular bot.tags.get("handling") as well